its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/ProjectKindTest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.it.scanner.msbuild.sonarqube;
+
+import com.sonar.it.scanner.msbuild.utils.ScannerClassifier;
+import com.sonar.it.scanner.msbuild.utils.TestUtils;
+import com.sonar.orchestrator.build.BuildResult;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.sonarqube.ws.Ce;
+import org.sonarqube.ws.Components;
+import org.sonarqube.ws.Issues.Issue;
+import org.sonarqube.ws.client.WsClient;
+import org.sonarqube.ws.client.components.ShowRequest;
+
+import static com.sonar.it.scanner.msbuild.sonarqube.Tests.ORCHESTRATOR;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+@ExtendWith(Tests.class)
+class ProjectKindTest {
```

Comment:
I'm not very happy with this name. Let me know if you have a better idea.

It's meant to test things that differ by different project types, like SDK/non-SDK/xaml/razor/aspire/whatever. And "ProjectType" in scanner context means Main/Test code, so we should not use that.

Reply:
SolutionKindTest?

Reply:
Sounds better